### DISCUSSION TITLE
Bulk Deletion In Lancedb

### DISCUSSION THREAD

**Discussion by Raghulkannan14** [lancedb/lancedb#1431] [Q&A]
I am using 

**Sample Code:-
tbl.delete("id in (1,2,3)")** 

like this for 20000 ids after 1000ids it is throwing error like unable to do deletion for ids, and also it is occupying the storage, How to do bulk deletion without an error

**Comment by wjones127** [comment]
Hello, it is difficult to debug your issue without an exact error message. Could you provide that?

The way you are deleting is the preferred way to delete rows.

**Reply by Raghulkannan14** [reply]
It something like concurrent deletion error, when i am trying to delete 20000 records and append the same 20000 records into the database by making minor update
